src/Illuminate/Redis/Connections/Connection.php

```diff
@@ -108,12 +115,31 @@ public function psubscribe($channels, Closure $callback)
      * @param  string  $method
      * @param  array  $parameters
      * @return mixed
+     *
+     * @throws \Exception Throws an exception if this connection should not be used
      */
     public function command($method, array $parameters = [])
     {
+        if ($this->shouldReconnect) {
+            // shouldReconnect is checked by RedisManager before returning this connection.
+            // If this exception is triggered, it means some other codepath is using this
+            // which should also be made aware of the public bool shouldReconnect.
+
+            // Basically - if you see this, this is a bug in whatever is NOT using RedisManager
+            // to get Redis objects.
+            throw new \Exception('The Redis connection '.$this->name.' has been marked as failed and needs to be recreated.');
+        }
         $start = microtime(true);
 
-        $result = $this->client->{$method}(...$parameters);
+        try {
+            $result = $this->client->{$method}(...$parameters);
+        } catch (\RedisException $e) {
```

Comment:
Can you import all classes?

Reply:
As per my reply, this is specific only to Phpredis, and this deliberately only handling \RedisException

Reply:
@driesvints this may be resolved simply by ALWAYS reconnecting redis here, rather than only on one specific edge case. Waiting for feedback below, which would simplify this to simply removing the `if`.

https://github.com/laravel/framework/blob/9.x/src/Illuminate/Redis/Connections/PhpRedisConnection.php#L534